liquidctl/driver/commander_core.py

```diff
@@ -43,50 +49,46 @@ class CommanderCore(UsbHidDriver):
     def initialize(self, **kwargs):
         """Initialize the device and get the fan modes."""
 
-        # INIT
-        self._send_command(_CMD_INIT)
+        with self._wake_device_context():
+            # Get Firmware
+            res = self._send_command(_CMD_GET_FIRMWARE)
+            fw_version = (res[3], res[4], res[5])
 
-        # Get Firmware
-        res = self._send_command(_CMD_GET_FIRMWARE)
-        fw_version = (res[3], res[4], res[5])
+            status = [('Firmware version', '{}.{}.{}'.format(*fw_version), '')]
 
-        status = [('Firmware version', '{}.{}.{}'.format(*fw_version), '')]
+            # Get LEDs per fan
+            raw_data = self._read_data(_MODE_LED_COUNT, _DATA_TYPE_LED_COUNT)
 
-        # Get LEDs per fan
-        self._send_command(_CMD_RESET)
-        self._send_command(_CMD_SET_MODE, [_MODE_DETECT_RGB])
-        res = self._send_command(_CMD_GET)
-        num_rgb = res[5]
-        for i in range(0, num_rgb):
-            connected = u16le_from(res, offset=6+i*4) == 2
-            num_leds = u16le_from(res, offset=8+i*4)
-            label = 'AIO LED count' if i == 0 else f'RGB port {i} LED count'
-            status += [(label, num_leds if connected else None, '')]
-
-        # Get what temp sensors are connected
-        for i, temp in enumerate(self._get_temps()):
-            connected = temp is not None
-            label = 'Water temperature sensor' if i == 0 else f'Temperature sensor {i}'
-            status += [(label, connected, '')]
+            num_devices = raw_data[0]
```

Comment:
I find `raw_data` slightly confusing for someone trying to quickly match the code to a capture, given that it isn't the raw response (which is what I would expect based on the name).

That said, I think you're right to strip out the boilerplate in `_read_data()`, so maybe just invert the names: use `res` in the callers (e.g. here) and `raw_data` inside `_read_data()`.